src/providers/wfs/oapif/qgsoapifprovider.cpp

```diff
@@ -954,26 +1096,30 @@ bool QgsOapifSharedData::computeServerFilter( QString &errorMsg )
   }
 
   const QgsExpression expr( mClientSideFilterExpression );
-  const auto rootNode = expr.rootNode();
-  if ( !rootNode )
-    return false;
-  mServerFilter = translateNodeToServer( rootNode, mFilterTranslationState, mClientSideFilterExpression );
-  if ( mFilterTranslationState == QgsOapifProvider::FilterTranslationState::PARTIAL )
-  {
-    QgsDebugMsg( QStringLiteral( "Part of the filter will be evaluated on client-side: %1" ).arg( mClientSideFilterExpression ) );
-  }
-  else if ( mFilterTranslationState == QgsOapifProvider::FilterTranslationState::FULLY_CLIENT )
+  bool ret = computeFilter( expr, mFilterTranslationState, mServerFilter, mClientSideFilterExpression );
+  if ( ret )
   {
-    QgsDebugMsg( "Whole filter will be evaluated on client-side" );
+    if ( mFilterTranslationState == QgsOapifProvider::FilterTranslationState::PARTIAL )
+    {
+      QgsDebugMsg( QStringLiteral( "Part of the filter will be evaluated on client-side: %1" ).arg( mClientSideFilterExpression ) );
```

Comment:
Should be QgsDebugMsgLevel(.... , 2) also line 1108

Reply:
done